addons/account/models/account_move.py

```diff
@@ -987,7 +987,7 @@ def _apply_cash_rounding(self, diff_balance, diff_amount_currency, cash_rounding
 
         others_lines = self.line_ids.filtered(lambda line: line.account_id.user_type_id.type not in ('receivable', 'payable'))
         others_lines -= existing_cash_rounding_line
-        total_amount_currency = sum(others_lines.mapped('amount_currency'))
+        total_amount_currency = self.currency_id.round(sum(others_lines.mapped('amount_currency')))
```

Comment:
It's probably for the best to round the closest where we need it. 
I would do it in the function `compute_difference` in https://github.com/odoo/odoo/blob/15.0/addons/account/models/account_cash_rounding.py#L53